src/Nethermind/Nethermind.Evm.Test/EvmPooledMemoryTests.cs

```diff
@@ -86,5 +87,13 @@ public void Load_should_update_size_of_memory()
             Assert.AreNotEqual(initialSize, memory.Size);
             Assert.AreEqual(expectedResult, result.ToArray());
         }
+
+        [Test]
+        public void GetTrace_should_not_thor_on_not_initialized_memory()
+        {
+            EvmPooledMemory memory = new();
+            memory.CalculateMemoryCost(0, 32);
```

Comment:
Is this needed? If you skip this, would is be enough as the scenario is uninitialized memory?

Reply:
Yes this is needed, because this will increase the `Size` of memory, while not actually increasing yet the actual `_memory` array. This is the root of the problem in previous `GetLogs` implementation.